Torch/TorchBase.cs

```diff
@@ -109,10 +117,19 @@ public bool IsOnGameThread()
             return Thread.CurrentThread.ManagedThreadId == MySandboxGame.Static.UpdateThread.ManagedThreadId;
         }
 
-        public async Task SaveGameAsync()
+        public async Task SaveGameAsync(Action<SaveGameStatus> callback)
         {
             Log.Info("Saving game");
-            if (MySandboxGame.IsGameReady && !MyAsyncSaving.InProgress && Sync.IsServer && !(MySession.Static.LocalCharacter?.IsDead ?? true))
+
+            if (!MySandboxGame.IsGameReady)
+            {
+                callback(SaveGameStatus.GameNotReady);
```

Comment:
Actions should be null checked in the beginning of the method or invoked like `action?.Invoke()` to avoid NREs when the action is null.